pkgs/top-level/ocaml-packages.nix

```diff
@@ -383,6 +399,29 @@ let
     ff-pbt = callPackage ../development/ocaml-modules/ff/pbt.nix { };
     ff-sig = callPackage ../development/ocaml-modules/ff/sig.nix { };
 
+    ffmpeg = callPackage ../development/ocaml-modules/ffmpeg { };
```

Comment:
You could use `callPackages` to make this more compact, see `all-packages.nix` for examples.

Reply:
I've now spent quite a few hours trying to accomplish this, but the end result seems extremely messy and overly complicated.

callPackages isn't really in the scope of the ocamlPackages set, so I'd have to access `pkgs.callPackages` which's scope probably doesn't include the required ocamlPackages. Accessing it with `pkgs.callPackagesWith ocamlPackages` probably wouldn't get me very far either as the question then becomes what do do with the other ocamlPackage sets...

In addition these ffmpeg packages are all really slightly different packages, which depend on each other. Most examples of callPackages seem to be building different versions of the package with basically the exact same build requirements.

My guess is that I'd need to do some magic with newScope here. But I think any such restructuring of the ocamlPackages subsystem is a little out of scope for this...

Maybe I'm missing something though, so if you have an obvious way of accomplishing this I'd be all ears. I don't think I fully understand callPackagesWith, newScope, and it's siblings.